### ISSUE TITLE
DataFrames does not precompile

### ISSUE THREAD

**Issue by IgorDouven** [JuliaData/DataFrames.jl#1408]
I did a Pkg.update(), which installed a new version of DataFrames (0.11.6+), but it won't precompile, giving the error message below. How can I resolve this?

```
julia> using DataFrames
INFO: Precompiling module DataFrames.
WARNING: Method definition ==(Base.Nullable{S}, Base.Nullable{T}) in module Base at nullable.jl:238 overwritten in module NullableArrays at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/v0.6/NullableArrays/src/operators.jl:99.
WARNING: Module DataFrames with uuid 9056792397081 is missing from the cache.
This may mean module DataFrames does not support precompilation but is imported by a module that does.
ERROR: LoadError: Declaring __precompile__(false) is not allowed in files that are being precompiled.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:455
 [2] require(::Symbol) at ./loading.jl:405
 [3] include_from_node1(::String) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
 [4] include(::String) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
 [5] anonymous at ./<missing>:2
while loading /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/v0.6/DataStreams/src/DataStreams.jl, in expression starting on line 248
ERROR: LoadError: LoadError: Failed to precompile DataStreams to /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/lib/v0.6/DataStreams.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:710
 [2] _require(::Symbol) at ./loading.jl:463
 [3] require(::Symbol) at ./loading.jl:405
 [4] include_from_node1(::String) at ./loading.jl:576
 [5] include(::String) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
 [6] include_from_node1(::String) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
 [7] include(::String) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:?
 [8] anonymous at ./<missing>:2
while loading /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/v0.6/DataFrames/src/abstractdataframe/io.jl, in expression starting on line 215
while loading /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/v0.6/DataFrames/src/DataFrames.jl, in expression starting on line 97
ERROR: Failed to precompile DataFrames to /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/lib/v0.6/DataFrames.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:710
 [2] _require(::Symbol) at ./loading.jl:497
 [3] require(::Symbol) at ./loading.jl:405
 [4] eval(::Module, ::Any) at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/julia/Contents/Resources/julia/lib/julia/sys.dylib:? (repeats 2 times)
 [5] macro expansion at /Applications/JuliaPro-0.6.2.2.app/Contents/Resources/pkgs-0.6.2.2/v0.6/Atom/src/repl.jl:117 [inlined]
 [6] anonymous at ./<missing>:?
```

**Comment by nalimilan** [comment]
It doesn't look like you're using 0.11.6+, as that version doesn't depend on NullableArrays. Or maybe one of the dependencies hasn't been updated as it should (e.g. CategoricalArrays)?

**Comment by nalimilan** [comment]
And what if you simply call `Pkg.update`? You probably have packages blocking the upgrade of DataFrames to 0.11. See https://discourse.julialang.org/t/how-to-upgrade-from-dataframes-0-10-1-to-0-11-3/8243/

**Comment by IgorDouven** [comment]
That's how I updated in the first place, just by Pkg.update()

2018-05-12 22:42 GMT+02:00 Milan Bouchet-Valat <notifications@github.com>:

**Comment by nalimilan** [comment]
That's not possible: `Pkg.update()` will never use 0.11.6+, only 0.11.6. Try calling `Pkg.free("DataFrames")`.

**Comment by IgorDouven** [comment]
Really, that's what I did (I have JuliaPro installed -- maybe that makes a difference).

Anyways, in the meantime I did a fresh install of JuliaPro, and so now I'm back to DataFrames 0.10.1 (meaning, among other things, that I can't use the MixedModels package as well as some other packages). I'm just going to wait now for the 0.7/1.0 release, hoping that by then the developers will have resolved these package incompatibilities.
